src/gdbstub_impl/ext/host_io.rs

```diff
@@ -31,25 +18,38 @@ impl<T: Target, C: Connection> GdbStubImpl<T, C> {
 
         crate::__dead_code_marker!("host_io", "impl");
 
+        macro_rules! handle_hostio_result {
+            ( if let Ok($val:pat) = $ret:expr => $callback:block ) => {{
+                match $ret {
+                    Ok($val) => $callback,
+                    Err(HostIoError::Errno(errno)) => {
+                        res.write_str("F-1,")?;
+                        res.write_num(errno as i32)?;
```

Comment:
remove the `as i32`. not sure why it was there to begin with...

Reply:
But `write_num` couldn't handle with `HostIoErrno`

Reply:
ah, true. in that case, this should be a u32 instead, as errno will never be negative